scripts/lib/CIME/namelist.py

```diff
@@ -171,6 +172,28 @@ def is_valid_fortran_name(string):
     """
     return FORTRAN_NAME_REGEX.search(string) is not None
 
+def get_fortran_name_only(varname):
+    """ remove array section if any and return only the variable name 
+    >>> get_fortran_name_only('foo')
+    'foo'
+    >>> get_fortran_name_only('foo(3)')
+    'foo'
+    >>> get_fortran_name_only('foo(::)')
+    'foo'
+    >>> get_fortran_name_only('foo(1::)')
+    'foo'
+    >>> get_fortran_name_only('foo(:+2:)')
+    'foo'
+    >>> get_fortran_name_only('foo(::-3)')
+    'foo'
+    >>> get_fortran_name_only('foo(::)')
+    'foo'
+    """
```

Comment:
Also, I don't see bounds error checking. It seems to happily accept negative bounds (e.g., `dtlimit(-3:2) = 0.0`).